.github/workflows/mcad-CI.yml

```diff
@@ -0,0 +1,40 @@
+# This workflow will build mcad and run unit test and E2E tests
+name: MCAD-CI
+on:
+  pull_request:
+    branches:
+      - 'main'
+
+jobs:
+  MCAD-CI:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.18.10
+
+      - name: Build
+        run:
+          make mcad-controller
+
+      - name: Run Unit tests
+        run: make run-test
+
+      - name: Build Image
```

Comment:
I think it will be a nice enhancement if we were to push images from PR(s). @sutaakar and @Srihari1192 are there any concerns with doing that? It would help with testing PR(s) on larger clusters.

Reply:
I think it is ok, can be done.

Reply:
so do we need to push images for each PR to some registry( quay or something) for testing purposes

Reply:
yes, AFAIK the same as it is done now with Travis

Reply:
Yeah understood , will implement similarly by using make command to push images